script/prerender/manage.ts

```diff
@@ -0,0 +1,397 @@
+      'ApproximateNumberOfMessagesDelayed',
+      'ApproximateNumberOfMessagesNotVisible',
+    ],
+    QueueUrl: workQueueUrl,
+  });
+  while (true) {
+    const getQueueAttributesResult = await sqsClient.send(getQueueAttributesCommand);
+    const attributes = assertDefined(
+      getQueueAttributesResult.Attributes,
+      '[SQS] [GetQueueAttributes] Unexpected response: missing Attributes key'
+    );
+
+    const numMessages = parseInt(attributes.ApproximateNumberOfMessages, 10) +
+      parseInt(attributes.ApproximateNumberOfMessagesDelayed, 10) +
+      parseInt(attributes.ApproximateNumberOfMessagesNotVisible, 10);
+
+    if (numMessages === 0) { break; }
+
+    console.log(`${numMessages}/${numJobs} prerendering jobs remaining`);
+
+    if (new Date() > timeoutDate) {
+      throw new Error(`Not all prerendering jobs finished within ${
+        PRERENDER_TIMEOUT_SECONDS} seconds of stack creation.`);
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  }
+
+  console.log(`All ${numJobs} prerendering jobs finished`);
+
+  console.log('Ensuring that the dead letter queue is empty');
+
+  // Ensure that the dead letter queue is also empty
+  // Since we are the only consumer of the dead letter queue, we use long polling to check
+  // that it is empty rather than waiting another minute and checking the queue attributes
+  const receiveDLQMessageResult = await sqsClient.send(new ReceiveMessageCommand({
+    MaxNumberOfMessages: 10,
+    QueueUrl: deadLetterQueueUrl,
+  }));
+
+  const dlqMessages = receiveDLQMessageResult.Messages || [];
+
+  if (dlqMessages.length > 0) {
+    throw new Error(`Some pages failed to render: ${JSON.stringify(dlqMessages)}`);
+  }
+
+  console.log(`The dead letter queue is empty; all ${numJobs} jobs finished successfully`);
+
+  // All pages, sitemaps and sitemap index have been rendered at this point
+}
+
+async function cleanup() {
+  // Proceed with other tasks while the stack deletes
+  await deleteWorkersStack();
+
+  await renderManifest();
+  await createRedirects(archiveLoader, osWebLoader);
+
+  const elapsedMinutes = globalMinuteCounter();
+
+  console.log(
+    `Prerender complete in ${elapsedMinutes} minutes. Rendered ${numPages} pages, ${
+    numBooks} sitemaps and the sitemap index. ${numPages / elapsedMinutes}ppm`
+  );
+
+  /* TODO?: Wait for the stack to delete and fail the build if it fails to delete
+            We can do it to get notified and prevent having a bunch of failed delete stacks,
+            but for the purposes of the build itself we are already done */
+}
+
+// Start creating the worker stack first since it takes a while
+// Do not wait for the stack creation to complete since we have other tasks to do first
+createWorkersStack().then(() => manage().catch(async(e) => {
+  // catch for manage() only
+  // Do not wait for the stack deletion to complete since we can't handle a delete failure anyway
+  await deleteWorkersStack();
```

Comment:
this should be in a finally so it doesn't need to be called in two places

Reply:
Moved to a try-finally with await instead of then().finally() since that's probably more readable.